libraries/AC_AutoTune/AC_AutoTune.cpp

```diff
@@ -485,15 +486,15 @@ void AC_AutoTune::control_attitude()
         // hold the copter level for 0.5 seconds before we begin a twitch
         // reset counter if we are no longer level
         if (!currently_level()) {
-            step_start_time = now;
+            step_start_time_ms = now + AUTOTUNE_REQUIRED_LEVEL_TIME_MS;
         }
 
         // if we have been level for a sufficient amount of time (0.5 seconds) move onto tuning step
-        if (now - step_start_time >= AUTOTUNE_REQUIRED_LEVEL_TIME_MS) {
+        if ((now > step_start_time_ms) || (now > step_time_limit_ms)) {
```

Comment:
It's an unlikely problem but I think this will cause problems if the clock wraps around back to zero..